packages/react-router5/modules/BaseLink.ts

```diff
@@ -1,22 +1,23 @@
 import { NavigationOptions, State, Router } from 'router5'
 import React, { Component, HTMLAttributes, MouseEventHandler } from 'react'
 
-export interface BaseLinkProps extends HTMLAttributes<HTMLAnchorElement> {
+export interface BaseLinkProps extends HTMLAttributes<HTMLElement> {
     routeName: string
     routeParams?: { [key: string]: any }
     routeOptions?: NavigationOptions
     className?: string
     activeClassName?: string
     activeStrict?: boolean
     ignoreQueryParams?: boolean
-    onClick?: MouseEventHandler<HTMLAnchorElement>
-    onMouseOver?: MouseEventHandler<HTMLAnchorElement>
+    onClick?: MouseEventHandler<HTMLElement>
+    onMouseOver?: MouseEventHandler<HTMLElement>
     successCallback?(state?: State): void
     errorCallback?(error?: any): void
     target?: string
     route?: State
     previousRoute?: State
     router: Router
+    component?: any
```

Comment:
I don't speak Typescript so I just left "any" here. This should have the type of "a string OR a function OR an instance of Component". I'm asking for help on this one.